src/staking/manager.ts

```diff
@@ -0,0 +1,660 @@
+    }
+    // Get the param based on the tip height
+    const p = getBbnParamByBtcHeight(tipHeight, this.stakingParams);
+    
+    const stakerInfo = await this.btcProvider.getStakerInfo();
+    if (!stakerInfo) {
+      throw new Error("Staker info not found while estimating staking fee");
+    }
+
+    const inputUTXOs = await this.btcProvider.getUTXOs();
+    if (inputUTXOs.length === 0) {
+      throw new Error("No input UTXOs provided");
+    }
+
+    const staking = new Staking(
+      this.network,
+      stakerInfo,
+      p,
+      stakingInput.finalityProviderPkNoCoordHex,
+      stakingInput.stakingTimelock,
+    );
+
+    const { fee: stakingFee } = staking.createStakingTransaction(
+      stakingInput.stakingAmountSat,
+      inputUTXOs,
+      feeRate,
+    );
+
+    return stakingFee;
+  }
+
+  /**
+   * Creates a signed staking transaction that is ready to be sent to the BTC 
+   * network.
+   * @param stakingInput - The staking inputs.
+   * @param unsignedStakingTx - The unsigned staking transaction.
+   * @param stakingParamVersion - The param version that was used to create the EOI
+   * @returns The signed staking transaction.
+   */
+  async createSignedBtcStakingTransaction(
+    stakingInput: StakingInputs,
+    unsignedStakingTx: Transaction,
+    stakingParamVersion: number,
+  ): Promise<Transaction> {
+    // Get the param based on version from the EOI
+    const p = getBbnParamByVersion(stakingParamVersion, this.stakingParams);
+    if (!p) {
+      throw new Error(`Unable to find staking params for version ${stakingParamVersion}`);
+    }
+    const stakerInfo = await this.btcProvider.getStakerInfo();
+    if (!stakerInfo) {
+      throw new Error("Staker info not found while creating signed staking transaction");
+    }
+    const inputUTXOs = await this.btcProvider.getUTXOs();
+    if (inputUTXOs.length === 0) {
+      throw new Error("No input UTXOs provided");
+    }
+
+    const staking = new Staking(
+      this.network,
+      stakerInfo,
+      p,
+      stakingInput.finalityProviderPkNoCoordHex,
+      stakingInput.stakingTimelock,
+    );
+
+    const stakingPsbt = staking.toStakingPsbt(
+      unsignedStakingTx,
+      inputUTXOs,
+    );
+
+    const signedStakingPsbtHex = await this.btcProvider.signPsbt(
+      stakingPsbt.toHex()
+    );
+    return Psbt.fromHex(signedStakingPsbtHex).extractTransaction();
+  }
+
+  /**
+   * Creates a signed unbonding transaction that is ready to be sent to the BTC 
+   * network.
+   * @param stakingInput - The staking inputs.
+   * @param stakingParamVersion - The param version that was used to create the 
+   * EOI
+   * @param stakingTx - The staking transaction.
+   * @param unsignedUnbondingTx - The unsigned unbonding transaction.
+   * @param covenantUnbondingSignatures - The covenant unbonding signatures.
+   * @returns The signed unbonding transaction.
+   */
+  async createSignedBtcUnbondingTransaction(
```

Comment:
Same here, while I find it useful for the SDK to provide methods for re-constructing a fully signed unbonding transaction, it doesn't seem right for the staker signature to be created inside the SDK instead of passed as a parameter.